src/streams/CreateStreamScreen.js

```diff
@@ -20,13 +22,19 @@ type Props = $ReadOnly<{|
 export default function CreateStreamScreen(props: Props): Node {
   const auth = useSelector(getAuth);
   const ownEmail = useSelector(getOwnEmail);
+  const getStreams = useSelector(getStreamsByName);
+  const allStreams = Array.from(getStreams.keys());
 
   const handleComplete = useCallback(
     (name: string, description: string, isPrivate: boolean) => {
-      api.createStream(auth, name, description, [ownEmail], isPrivate);
-      NavigationService.dispatch(navigateBack());
+      if (allStreams.includes(name)) {
+        showErrorAlert('A Stream with this name already exists');
+      } else {
+        api.createStream(auth, name, description, [ownEmail], isPrivate);
+        NavigationService.dispatch(navigateBack());
+      }
```

Comment:
Instead of putting the normal, happy-case code in an `else`, let's use the "early return" pattern here. So, in the `if`, just after the `showErrorAlert` line, add a `return` line. Then pull what's currently in the `else` block outward. Like this:

```js
      if (allStreams.includes(name)) {
        showErrorAlert('A Stream with this name already exists');
        return;
      }

      api.createStream(auth, name, description, [ownEmail], isPrivate);
      NavigationService.dispatch(navigateBack());
```

For Greg's reasoning about this preference, see https://github.com/zulip/zulip-mobile/pull/4980#discussion_r700651438 .